common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
maybe `SingleThreadedEventExecutor` should just pass the remaining time to `afterRunningAllTasks`?